src/network/taraxa_capability.cpp

```diff
@@ -762,112 +762,74 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
         break;
       }
 
-      auto pbft_blk_count = _r.itemCount();
-      LOG(log_dg_pbft_sync_) << "PbftBlockPacket received, num of pbft blocks to be processed: " << pbft_blk_count;
+      auto itemCount = _r.itemCount();
+      if (itemCount > 0) {
+        auto it = _r.begin();
+        bool last_block = (*it++).toInt<bool>();
+        PbftBlockCert pbft_blk_cert(*it);
+        LOG(log_nf_pbft_sync_) << "PbftBlockPacket received. Period: " << pbft_blk_cert.pbft_blk->getPeriod();
 
-      auto pbft_sync_period = pbft_chain_->pbftSyncingPeriod();
-      for (auto const pbft_blk_tuple : _r) {
-        dev::RLP dag_blocks_rlp = pbft_blk_tuple[0];
         string received_dag_blocks_str;
-        map<uint64_t, map<blk_hash_t, pair<DagBlock, vector<Transaction>>>> dag_blocks_per_level;
-        for (auto const dag_blk_struct : dag_blocks_rlp) {
-          DagBlock dag_blk(dag_blk_struct[0]);
-          auto const &dag_blk_h = dag_blk.getHash();
-          peer->markBlockAsKnown(dag_blk_h);
-          std::vector<Transaction> new_transactions;
-          for (auto const trx_raw : dag_blk_struct[1]) {
-            auto &trx = new_transactions.emplace_back(trx_raw);
-            peer->markTransactionAsKnown(trx.getHash());
-          }
-          received_dag_blocks_str += dag_blk_h.toString() + " ";
-          auto level = dag_blk.getLevel();
-          dag_blocks_per_level[level][dag_blk_h] = {move(dag_blk), move(new_transactions)};
-        }
-        LOG(log_dg_dag_sync_) << "Received Dag Blocks: " << received_dag_blocks_str;
-        for (auto const &block_level : dag_blocks_per_level) {
+        for (auto const &block_level : pbft_blk_cert.dag_blocks_per_level) {
           for (auto const &block : block_level.second) {
-            auto status = checkDagBlockValidation(block.second.first);
-            if (!status.first) {
-              LOG(log_si_pbft_sync_) << "PBFT SYNC ERROR, DAG missing a tip/pivot"
-                                     << ", PBFT chain size: " << pbft_chain_->getPbftChainSize()
-                                     << ", synced queue size: " << pbft_chain_->pbftSyncedQueueSize();
-              syncing_state_.set_peer_malicious();
-              host->disconnect(_nodeID, p2p::UserReason);
-              restartSyncingPbft(true);
-              return;
+            received_dag_blocks_str += block.getHash().toString() + " ";
+            if (block.getLevel() > peer->dag_level_) {
```

Comment:
as `pbft_blk_cert.dag_blocks_per_level` is ordered map, last item will have highest level, no need to check in loop